gear/form_qc_checker/src/python/form_qc_app/datastore.py

```diff
@@ -269,19 +336,47 @@ def get_previous_nonempty_record(
                 file_name=visit["file.name"], acq_id=visit["file.parents.acquisition"]
             )
 
-            if not visit_data:
-                continue
+            if self.__check_nonempty(ignore_empty_fields, visit_data):
+                return visit_data
+
+        log.warning(
+            "No previous visit found with non-empty values for %s", ignore_empty_fields
+        )
+        return None
 
-            found_all = True
-            for field in fields:
-                if not visit_data.get(field):
-                    found_all = False
-                    break
+    def get_initial_record(
+        self,
+        current_record: Dict[str, Any],
+        ignore_empty_fields: Optional[List[str]] = None
+    ) -> Optional[Dict[str, Any]]:
+        """Overriding the abstract method, get the initial visit record for the
+        specified participant if non-empty.
 
-            if found_all:
-                return visit_data
+        Args:
+            current_record: record currently being validated
+            ignore_empty_fields: Field(s) to check for blanks
+
+        Returns:
+            dict[str, str]: initial visit record if found and non-empty,
+                            None otherwise
+        """
+        # will be None if we've never looked for it, and empty dict if
+        # we tried looking for it but it could not be found
+        if self.__initial_visit is None:
+            self.__initial_visit = self.__get_initial_visit(current_record)
+
+        if not self.__initial_visit:
+            return None
+
+        visit_data = self.__forms_store.get_visit_data(
+            file_name=self.__initial_visit["file.name"],
+            acq_id=self.__initial_visit["file.parents.acquisition"],
+        )
+
+        if self.__check_nonempty(ignore_empty_fields, visit_data):
+            return visit_data
 
-        log.warning("No previous visit found with non-empty values for %s", fields)
+        log.warning("Initial visit has non-empty values for %s", ignore_empty_fields)
```

Comment:
If we want these fields to be non-empty in the IVP packet, this warning needs to be modified as "No initial visit found with non-empty values for %s"